tests/unit/dataset.spec.ts

```diff
@@ -135,3 +140,141 @@ describe('_guessType', () => {
+      data: [
+        ['Paris', 10000000, true],
+        ['Marseille', 3000000, false],
+        ['Berlin', 1300000, true],
+        ['Avignon', 100000, false],
+        ['La Souterraine', 0, false],
+        ['New York City', 10000000, false],
+        ['Rio de Janeiro', 4000000, false],
+        [undefined, null, 10],
+      ],
+    };
+    const datasetWithInferredType = inferTypeFromDataset(dataset, 7);
+    expect(datasetWithInferredType.headers).to.eql([
+      { name: 'city', type: 'string' },
+      { name: 'population', type: 'integer' },
+      { name: 'isCapitalCity', type: 'boolean' },
+    ]);
+  });
+
+  it('should not infer type with mixed values', () => {
+    const dataset: DataSet = {
+      headers: [{ name: 'city' }, { name: 'population' }, { name: 'isCapitalCity' }],
+      data: [
+        ['Paris', 10000000, true],
+        ['Marseille', 3000000, false],
+        ['Berlin', 1300000, true],
+        ['Avignon', 100000, false],
+        ['La Souterraine', 0, false],
+        ['New York City', 10000000, false],
+        ['Rio de Janeiro', 4000000, false],
+        [undefined, false, 10],
+      ],
+    };
+
+    const datasetWithInferredType = inferTypeFromDataset(dataset);
+    expect(datasetWithInferredType.headers).to.eql([
+      { name: 'city' },
+      { name: 'population' },
+      { name: 'isCapitalCity' },
+    ]);
+  });
+
+  it('should not infer type on column with undefined values', () => {
+    const dataset: DataSet = {
+      headers: [{ name: 'city' }, { name: 'population' }, { name: 'isCapitalCity' }],
+      data: [['Paris', 10000000, undefined]],
+    };
+
+    const datasetWithInferredType = inferTypeFromDataset(dataset, 1);
+    expect(datasetWithInferredType.headers).to.eql([
+      { name: 'city', type: 'string' },
+      { name: 'population', type: 'integer' },
+      { name: 'isCapitalCity' },
+    ]);
+  });
+
+  it('should not infer type on column with null values', () => {
+    const dataset: DataSet = {
+      headers: [{ name: 'city' }, { name: 'population' }, { name: 'isCapitalCity' }],
+      data: [['Paris', 10000000, null]],
+    };
+
+    const datasetWithInferredType = inferTypeFromDataset(dataset, 1);
+    expect(datasetWithInferredType.headers).to.eql([
+      { name: 'city', type: 'string' },
+      { name: 'population', type: 'integer' },
+      { name: 'isCapitalCity' },
+    ]);
+  });
+
+  it("should infer type on column with null values if there's other values in the column", () => {
+    const dataset: DataSet = {
+      headers: [{ name: 'city' }, { name: 'population' }, { name: 'isCapitalCity' }],
+      data: [['Paris', 10000000, null], ['Paris', 10000000, false]],
+    };
+
+    const datasetWithInferredType = inferTypeFromDataset(dataset, 1);
+    expect(datasetWithInferredType.headers).to.eql([
+      { name: 'city', type: 'string' },
+      { name: 'population', type: 'integer' },
+      { name: 'isCapitalCity' },
```

Comment:
Hmm, shouldn't it be `{ name: 'isCapitalCity', type: 'boolean' }`?

Reply:
You got it right, the test pass because I let `maxRows` set to 1